wandb-scim/custom_roles.py

```diff
@@ -0,0 +1,245 @@
+            "Content-Type": "application/json"
+        }
+
+        response = requests.patch(url, json=data, headers=headers)
+
+        if response.status_code == 200:
+            updated_data = response.json()  # Get the updated resource data from the response
+            print("Updated Data:", updated_data)
+            return "Custom role updated successfully"
+
+        elif response.status_code == 404:
+            return "Custom role not found"
+        else:
+            return f"Failed to update custom role. Status code: {response.status_code}"
+
+    def remove_permission(self, role_id, request_payload):
+        """
+        Removes permission from a custom role.
+
+        Args:
+            role_id (str): role_id from the custom role.
+            request_payload (dict): The payload containing permission information.
+                It should contain the following key:
+                    - 'permissionJson': The permissions JSON to be removed from the custom role.
+
+        Returns:
+            str: A message indicating whether the permission removal was successful or failed.
+        """
+        print("Remove permission from a custom role")
+        data = {
+            "schemas": ["urn:ietf:params:scim:api:messages:2.0:PatchOp"],
+            "Operations": [
+                {
+                    "op": "remove",
+                    "path": "permissions",
+                    "value": [request_payload['permissionJson']]
+                }
+            ]
+        }
+        headers = {
+            "Authorization": self.authorization_header,
+            "Content-Type": "application/json"
+        }
+        url = f"{self.base_url}/scim/Roles/{role_id}"
+        response = requests.patch(url, json=data, headers=headers)
+
+        if response.status_code == 200:
+            updated_data = response.json()  # Get the updated resource data from the response
+            print("Updated Data:", updated_data)
+            return "Custom role updated successfully"
+
+        elif response.status_code == 404:
+            return "Custom role not found"
+        else:
+            return f"Failed to update custom role. Status code: {response.status_code}"
+
+    def update_custom_role(self, role_id, request_payload):
+        """
+        Updates name and description of a custom role.
+
+        Args:
+            role_id (str): role_id from the custom role.
+            request_payload (dict): The payload containing role information.
+                It should contain the following keys:
+                    - 'roleName': The name of the custom role.
+                    - 'roleDescription': The description of the custom role.
+                    - 'inheritedFrom': The inheritance information for the custom role.
+
+        Returns:
+            str: A message indicating whether the custom role update was successful or failed.
+        """
+        print("Update name and description of custom role")
+        data = {
```

Comment:
Repeating the feedback here - we would like to update this conditionally for every field based on if it's provided in the input request. All three fields -  name, description and inheritedFrom are optional here, so one could specify just one of those, or two of those, or all three to be updated. So we should construct the SCIM request field by field only if it's available in the payload.